src/coreclr/src/jit/compiler.cpp

```diff
@@ -3139,6 +3092,14 @@ void Compiler::compInitOptions(JitFlags* jitFlags)
     }
 #endif // FEATURE_FASTTAILCALL
 
+#ifdef CONFIGURABLE_ARM_ABI
+    opts.compUseSoftFP = !!JitConfig.JitSoftFP();
+    if (opts.compUseSoftFP)
+    {
+        GlobalJitOptions::compFeatureHfa = !opts.compUseSoftFP;
```

Comment:
I'm not sure we want this as a static global. It seems like it needs to be a member of the `Compiler` object. What if we want to allow compiling both for arm and armel in the same process? Or if multiple threads try to initialize this concurrently?

Also, I don't think the config variable JitSoftFP should be the way this ABI is communicated to the JIT; it should come in via the jit flags, as a CorJitFlag.

Reply:
The problem with compFeatureHfa is that it needs to be useable from the code in the code in LclVarDsc and fgArgTabEntry structures which don't have access to a Compiler object to query. Of course, I could pass the Compiler* as a parameter to these functions, but that didn't seem like a good approach. Given that the current model for usage of this is to only support one target architecture per process, I just went with a static. An alternative approach would be to use a threadstatic, which would allow for different compilations in the same process to have different values.

---
In reply to: [510494195](https://github.com/dotnet/runtime/pull/43706#discussion_r510494195) [](ancestors = 510494195)

Reply:
We already stash `Compiler*` in a thread static: `JitTls::GetCompiler()`, but I'm not sure it's used in non-DEBUG code.